BigFileMergeSort/src/ua/goit/alg/BigFileMergeSort.java

```diff
@@ -0,0 +1,182 @@
+package ua.goit.alg;
+
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class BigFileMergeSort {
+	static final String property = "java.io.tmpdir";
+	static String tempDir = System.getProperty(property);
+
+	public static File mergeSort(File file, int blocksize) throws IOException {
+		ArrayList<File> partList = splitSort(file, blocksize);
+		File result = File.createTempFile("result", ".txt", new File(tempDir));
+		result.deleteOnExit(); 
+		result = mergeParts(partList).get(0);
+		return result;
+	}
+
+	private static ArrayList<File> mergeParts(ArrayList<File> partList) throws IOException {
+		ArrayList<File> result = null;	
+		if (partList.size() <= 1) {
+			result = partList;
+		} else {
+			result =  mergePartsProcess(partList);
+		} 
+		return result;
+	}
+
+	private static ArrayList<File> mergePartsProcess(ArrayList<File> partList)
+			throws IOException {
+		ArrayList<File> resultList = new ArrayList<File>();
+		if (partList.size() % 2 == 1) {
+			File merged = mergeFiles(partList.get(0), partList.get(1));
+			partList.add(merged);
+			partList.remove(1);
+			partList.remove(0);
+		}
+		for (int i = 0; i < partList.size()-1; i+=2) {
+			resultList.add(mergeFiles(partList.get(i), partList.get(i+1)));
+		}
+		resultList = mergeParts(resultList);
```

Comment:
mergePartsProcess --> mergeParts--> mergePartsProcess, it's recursive, we don't need that here, let's do it iteratively